### PR TITLE
Version 1.1

### PR DESCRIPTION
**Pull request by russhwolf**
This branch contains changes that will be released as version 1.1. It includes fixes for issues #81 (superceding PR #117), #142, #157, #160, and #162.
 
TODO:
- [x] Sample project updates for 1.1
- [x] Readme updates
- [x] Docs for new serialization functions
- [x] Update remaining dependency versions

### CODE REVIEW

File: multiplatform-settings/src/appleMain/kotlin/com/russhwolf/settings/KeychainSettings.kt [hunk 1]
```diff
@@ -91,11 +96,16 @@ import platform.darwin.OSStatus
  * every key, if the default behavior does not fit your needs.
  */
 @ExperimentalSettingsImplementation
-public class KeychainSettings(vararg defaultProperties: Pair<CFStringRef?, CFTypeRef?>) : Settings {
+public class KeychainSettings @ExperimentalSettingsApi constructor(vararg defaultProperties: Pair<CFStringRef?, CFTypeRef?>) :
```
Comment (russhwolf) [inline multiplatform-settings/src/appleMain/kotlin/com/russhwolf/settings/KeychainSettings.kt:99]: I'm a little ambivalent about how I want to handle the fact that we directly expose a `@ExperimentalForeignApi`-annotated type in this constructor. (I think this is the only place in the codebase where that happens in public API, unless there are others I missed). I think the natural thing to do would be to re-apply the annotation, but it's not applicable to constructors. So I landed on marking it `@ExperimentalSettingsApi` instead. This lets the rest of the class go stable without depending on Kotlin stabilizing these APIs.

File: multiplatform-settings-serialization/src/commonMain/kotlin/SettingsSerialization.kt [hunk 2]
```diff
@@ -154,7 +152,32 @@ public fun <T> Settings.decodeValueOrNull(
...
+    serializer.deserialize(enumerator)
+    enumerator.removeKeys()
+}
+
+@ExperimentalSerializationApi
+@ExperimentalSettingsApi
+public fun <T> Settings.containsValue(
+    serializer: KSerializer<T>,
+    key: String,
+    serializersModule: SerializersModule = EmptySerializersModule()
```
Comment (vanniktech) [inline multiplatform-settings-serialization/src/commonMain/kotlin/SettingsSerialization.kt:179]:
```suggestion
    serializersModule: SerializersModule = EmptySerializersModule(),
```

File: multiplatform-settings/src/appleMain/kotlin/com/russhwolf/settings/KeychainSettings.kt [hunk 3]
```diff
@@ -14,10 +14,15 @@
  * limitations under the License.
  */
 
-@file:OptIn(UnsafeNumber::class) // due to CFIndex usage
+@file:OptIn(
+    UnsafeNumber::class, // due to CFIndex usage
```
Comment (vanniktech) [inline multiplatform-settings/src/appleMain/kotlin/com/russhwolf/settings/KeychainSettings.kt:18]:
```suggestion
    UnsafeNumber::class, // Due to CFIndex usage.
```

File: tests/build.gradle.kts [hunk 4]
```diff
@@ -27,30 +27,26 @@ kotlin {
             dependencies {
                 implementation(project(":multiplatform-settings"))
 
-                implementation(kotlin("test-common"))
-                implementation(kotlin("test-annotations-common"))
+                implementation(kotlin("test"))
```
Comment (vanniktech) [inline tests/build.gradle.kts:30]: TIL!

**Review by vanniktech — APPROVED** [review]
👌

File: multiplatform-settings-serialization/src/commonMain/kotlin/SettingsSerialization.kt (hunk 2 above)
Comment (russhwolf) [reply multiplatform-settings-serialization/src/commonMain/kotlin/SettingsSerialization.kt:234]: Maybe an unpopular opinion but I'm meh on trailing commas

**Comment by russhwolf** [comment]
Sample app build failure is expected right now. Updates in-progress in there.

**Pull request merged**